protos/backend_service.proto

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+syntax = "proto3";
+
+package durabletask.protos.backend.v1;
+
+option csharp_namespace = "Microsoft.DurableTask.Protobuf";
+option java_package = "com.microsoft.durabletask.implementation.protobuf";
+option go_package = "/internal/protos";
+
+import "orchestrator_service.proto";
+
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/empty.proto";
+
+// gRPC service used by Durable Task Framework (DTFx) backend implementations.
+// The RPCs in this service are used by DTFx clients to manage orchestrations.
+service ClientBackendService {
+    // Creates a new orchestration instance.
+    rpc CreateOrchestrationInstance (CreateInstanceRequest) returns (CreateInstanceResponse);
+
+    // Sends an event to an orchestration instance. This RPC is used for raising external events to orchestrations
+    // and for sending orchestration lifecycle events, such as terminate, suspend, resume, etc.
+    rpc AddOrchestrationEvent (AddOrchestrationEventRequest) returns (google.protobuf.Empty);
+
+    // Returns metadata about an orchestration instance.
+    rpc GetOrchestrationMetadata (GetInstanceRequest) returns (GetInstanceResponse);
+
+    // Waits for an orchestration to reach a terminal state and then returns metadata for that orchestration.
+    rpc WaitForOrchestrationCompletion(GetInstanceRequest) returns (GetInstanceResponse);
```

Comment:
Should have dedicated messages, which can allow extending in the future

Reply:
I'm following the convention (which might not be ideal) from orchestration_service.proto, which has a similar service contract. We can update these if/when we need to extend in the future.

Reply:
Changed my mind, based on my other realization about managing breaking changes.